test-menus.py

```diff
@@ -0,0 +1,94 @@
+import unittest
+from mock import patch
+from decorators import get_menu_entries
+
+class MenuTestCase(unittest.TestCase):
+
+    @patch('mod_auth.models.User')
+    def test_simple_menu(self,mock_user):
+        """
+        Passing a menu entry to get_menu_entries() when all the routes are accessible (simulating admin)
+        and verifying against correct menu structure.
+        """
+        mu=mock_user.return_value
+        mu.can_access_route.return_value=True
+
+        entries = get_menu_entries(
+            mu, 'Configuration', 'cog', '', [
+                {'title': 'Notif. services', 'icon': 'bell-o', 'route':
+                    'config.notifications', 'entries': [{'title': 'Notif. services', 'icon': 'bell-o', 'route':
+                    'config.notifications', 'entries': [{'title': 'Honeypot services', 'icon': 'sliders', 'route':
+                    'config.services', 'entries': [{'title': 'Profile mgmt', 'icon': 'bookmark', 'route':
+                    'honeypot.profiles'}]}]}]},
+                {'title': 'Data processing', 'icon': 'exchange', 'route':
+                    'config.data_processing'},
+                {'title': 'Honeypot services', 'icon': 'sliders', 'route':
+                    'config.services'}
+            ]
+        )
+
+        correct_entries={'entries': [{'title': 'Notif. services', 'route': 'config.notifications', 'entries': [{'title': 'Notif. services', 'route': 'config.notifications', 'entries': [{'title': 'Honeypot services', 'route': 'config.services', 'entries': [{'title': 'Profile mgmt', 'route': 'honeypot.profiles', 'icon': 'bookmark'}], 'icon': 'sliders'}], 'icon': 'bell-o'}], 'icon': 'bell-o'}, {'route': 'config.data_processing', 'title': 'Data processing', 'icon': 'exchange'}, {'route': 'config.services', 'title': 'Honeypot services', 'icon': 'sliders'}], 'icon': 'cog', 'title': 'Configuration'}
```

Comment:
Try to reformat this over multiple lines so it's easier to read :+1:

Reply:
Done in the latest commit.